src/extract.ts

```diff
@@ -0,0 +1,48 @@
+import os from 'os'
+import { promises as fsp, existsSync } from 'fs'
+import { resolveModuleExportNames } from 'mlly'
+import { resolvePackageJSON, readPackageJSON } from 'pkg-types'
+import { dirname, join } from 'pathe'
+import { Import, PackagePreset } from './types'
+
+export async function resolvePackagePreset (preset: PackagePreset): Promise<Import[]> {
+  const scanned: string[] = await extractExports(preset.package, preset.url, preset.cache)
+
+  const filtered = scanned.filter((name) => {
+    for (const item of preset.ignore || []) {
+      if (typeof item === 'string' && item === name) {
+        return false
+      }
+      if (item instanceof RegExp && item.test(name)) {
+        return false
+      }
+      if (typeof item === 'function' && item(name) === false) {
+        return false
+      }
+    }
+    return true
+  })
+
+  return filtered.map(name => (<Import>{
+    from: preset.package,
+    name
+  }))
+}
+async function extractExports (name: string, url?: string, cache = true) {
+  const packageJsonPath = await resolvePackageJSON(name, { url })
+  const packageJson = await readPackageJSON(packageJsonPath)
+  const version = packageJson.version
```

Comment:
This is rare and shouldn't happen but private packages can omit version 
```suggestion
  const version = packageJson.version ||  '0.0.0'
```

Reply:
Updated the impl, I guess it's better to skip the cache for private packages as they are addressed to change right?